src/qt/psbtoperationsdialog.cpp

```diff
@@ -46,20 +46,23 @@ PSBTOperationsDialog::~PSBTOperationsDialog()
 void PSBTOperationsDialog::openWithPSBT(PartiallySignedTransaction psbtx)
 {
     m_transaction_data = psbtx;
-
-    bool complete;
-    size_t n_could_sign;
-    FinalizePSBT(psbtx);  // Make sure all existing signatures are fully combined before checking for completeness.
-    TransactionError err = m_wallet_model->wallet().fillPSBT(SIGHASH_ALL, false /* sign */, true /* bip32derivs */, &n_could_sign, m_transaction_data, complete);
-    if (err != TransactionError::OK) {
-        showStatus(tr("Failed to load transaction: %1")
-            .arg(QString::fromStdString(TransactionErrorString(err).translated)), StatusLevel::ERR);
-        return;
+    if (m_wallet_model) {
+        bool complete;
+        size_t n_could_sign;
+        FinalizePSBT(psbtx); // Make sure all existing signatures are fully combined before checking for completeness.
+        TransactionError err = m_wallet_model->wallet().fillPSBT(SIGHASH_ALL, false /* sign */, true /* bip32derivs */, &n_could_sign, m_transaction_data, complete);
+        if (err != TransactionError::OK) {
+            showStatus(tr("Failed to load transaction: %1")
+                           .arg(QString::fromStdString(TransactionErrorString(err).translated)),
+                       StatusLevel::ERR);
+            return;
+        }
+        m_ui->broadcastTransactionButton->setEnabled(complete);
+        m_ui->signTransactionButton->setEnabled(!complete && !m_wallet_model->wallet().privateKeysDisabled() && n_could_sign > 0);
+    } else {
+        m_ui->broadcastTransactionButton->setEnabled(true);
```

Comment:
It's not always correct for the broadcast button to be enabled since the user could provide a PSBT that is incomplete. By always doing `FinalizePSBT` as I have suggested above, then we will know whether the PSBT is complete by the time we get here.